game/src/graphic/textures/TextureHandler.java

```diff
@@ -2,30 +2,63 @@
 
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.files.FileHandle;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Arrays;
+import java.util.Comparator;
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Predicate;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 /**
  * A texture handler class for managing textures paths for further use.
  *
- * <p>Singleton (possibly thread-safe).
+ * <p>This class is designed as Singleton, because all asset paths should be read only once at the
+ * beginning of the application.
  */
 public class TextureHandler {
+    public static final String PLACEHOLDER_FILENAME = ".resource_root";
+    private static final int MAX_DEPTH_GUESS = 4;
     private static final TextureHandler INSTANCE = new TextureHandler();
 
     private final Map<String, Set<FileHandle>> pathMap = new LinkedHashMap<>();
 
     private TextureHandler() {
-        addAllAssets(Gdx.files.internal(Gdx.files.getLocalStoragePath()));
+        try {
+            addAllAssets(getResourceRoot());
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.out.println("No resource root found.");
+            System.exit(0);
+        }
```

Comment:
wieso machst du jetzt hier try/catch? 

und wenn das sinnvoll ist: wir arbeiten mit loggern - bitte keine stacktraces oder system.out.println ...

und sollte ein system.exit nicht weiter "oben" erfolgrn?

Reply:
system.exit beendet alles, die Ausgabe muss vorher erfolgen ...
die Exception kann bei Files.walk auftreten

Reply:
das ist mir klar, java verstehe ich.

was mir nicht klar ist: warum tust du das hier?!

Reply:
> was mir nicht klar ist: warum tust du das hier?!

weil das Spiel dann in einem Zustand ist, mit dem es nicht gespielt werden kann(?)

Reply:
ja, aber warum steigst du hier einseitig aus? 

die exception konnte vor deinem fix auch passieren, dennoch wurde das spiel und die jvm nicht abgeschossen. also, warum führst du das jetzt hier mit dem fix ein?

das beenden sollte nur in game stattfinden und nicht in irgendwelchen helferklassen.

Reply:
nagut, stimmt...

Reply:
habe `Gdx.app.exit();` wieder entfernt... sodass das Spiel dann zunächst weiterläuft